src/parcel/OutputTemplate.js

```diff
@@ -19,14 +19,46 @@ function helix_wrap_action(main) {
   const { pipe } = require('MOD_PIPE');
   const { pre, before, after } = require('MOD_PRE');
 
+  // todo: mode to helix-pipeline
+  const CONTEXT_PROPS = ['error', 'request', 'content', 'response'];
+  const CONTENT_PROPS = ['sources', 'body', 'mdast', 'sections', 'document', 'htast' ,'json', 'xml', 'meta', 'title', 'intro', 'image'];
+  const REQUEST_PROPS =  ['url', 'path', 'pathInfo', 'rootPath', 'selector', 'extension', 'method', 'headers', 'params'];
+  const RESPONSE_PROPS = ['status', 'body', 'hast', 'headers'];
+
+  const filterObject = (obj, allowedProperties) => {
+    if (!obj) {
+      return;
+    }
+    Object.keys(obj).forEach((key) => {
+      if (allowedProperties.indexOf(key) < 0) {
+        delete obj[key];
+      }
+    })
+  };
+
+  const sanitizeContext = (context) => {
+    filterObject(context, CONTEXT_PROPS);
+    filterObject(context.content, CONTENT_PROPS);
+    filterObject(context.request, REQUEST_PROPS);
+    filterObject(context.response, RESPONSE_PROPS);
+  };
+
   // this gets called by openwhisk
   return async function wrapped(params) {
     // this is the once function that will be installed in the pipeline
-    async function once(payload, action) {
+    async function once(context, action) {
       // calls the pre function and then the script's main.
       async function invoker(next) {
-        const ret = await Promise.resolve(pre(payload, action));
-        return Promise.resolve(next(ret || payload, action));
+        const ret = await Promise.resolve(pre(context, action));
+        const res = await Promise.resolve(next(ret || context, action));
+        if (res && res.response && res.response.body) {
+          if (!context.response) {
+            context.response = {};
+          }
+          context.response.body = res.response.body;
+        }
+        sanitizeContext(context);
```

Comment:
I'm not sure why you are doing this here. If we don't filter the context, the validator in the pipeline would complain anyway, so it can't be for protection against invalid properties?

Reply:
the problem is the following:

for feeding the HTL with data provided by the `pre.js`, they need to add properties to `context`. (this will become the `payload` object in HTL...). until now, we never returned the modified `context` from the `pre.js` back to the pipeline, only the result of the HTL.
since the `cont()` had a return value, it was merged with the `context` that was present before the `cont()`.

now, if we abandon the merge behaviour, we need to keep the `context` that was modified in the `pre.js`. as mentioned in the issue, the proper solution would be to offer a `context.data` property, that could be used for anything. 

but for now, we just remove all _garbage_ from the context, before continuing the pipeline. otherwise the validator would complain.